src/common/actions/snaps.js

```diff
@@ -14,85 +11,37 @@ export const REMOVE_SNAP_ERROR = 'REMOVE_SNAP_ERROR';
 
 export function fetchUserSnaps(owner) {
   return async (dispatch) => {
-    const url = `${BASE_URL}/api/launchpad/snaps/list`;
     const query = `owner=${encodeURIComponent(owner)}`;
 
     dispatch({
-      type: FETCH_SNAPS
-    });
-
-    try {
-      const response = await fetch(`${url}?${query}`, {
-        headers: { 'Content-Type': 'application/json' },
-        credentials: 'same-origin'
-      });
-      await checkStatus(response);
-      const result = await response.json();
-      if (result.status !== 'success') {
-        throw getError(response, result);
+      type: FETCH_SNAPS,
+      [CALL_API]: {
+        types: [ FETCH_SNAPS_SUCCESS, FETCH_SNAPS_ERROR ],
```

Comment:
It's the [same abstraction](https://github.com/reactjs/redux/blob/master/examples/real-world/src/actions/index.js) we discussed in IRC last week, so I'm not sure why it's confusing.

The principle benefits I see are that we can memoise settings such as the API base URL and the CSRF token when the middleware is added to Redux. A helper function would have to accept the settings by another route:

- By parameter, on every call.
- From a global.
- From a singleton pattern.

These trade-offs are undesirable, in my opinion. I think the middleware is a great option that follows an example set by the Redux team.